x-pack/plugins/maps/public/components/toolbar_overlay/set_view_control/set_view_control.js

```diff
@@ -4,23 +4,54 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import {
   EuiForm,
   EuiFormRow,
   EuiButton,
   EuiFieldNumber,
+  EuiButtonIcon,
+  EuiPopover,
 } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
 
-export class SetView  extends React.Component {
+function getViewString(lat, lon, zoom) {
+  return `${lat},${lon},${zoom}`;
+}
+
+export class SetViewControl extends Component {
 
   state = {
     lat: this.props.center.lat,
     lon: this.props.center.lon,
     zoom: this.props.zoom,
+    prevView: getViewString(this.props.center.lat, this.props.center.lon, this.props.zoom)
```

Comment:
We shouldn't be initializing state using props. I see this pattern was already in place but ideally we should just change all of these to their respective 0 or empty string values

Reply:
why not? That is very common react pattern. The props provide the initial values but then users can enter the values they want

Reply:
Resolved offline. Updated to use `getDerivedStateFromProps`